spec/datadog/core/telemetry/metric_spec.rb

```diff
@@ -0,0 +1,184 @@
+require 'spec_helper'
+
+require 'datadog/core/telemetry/metric'
+
+RSpec.describe Datadog::Core::Telemetry::Metric::Count do
+  subject(:metric) { described_class.new('tests', { foo: :bar }) }
+  let(:generate_metric_type) { 'generate-metrics' }
+
+  describe '.request_type' do
+    subject(:request_type) { metric.class.request_type }
+
+    it { is_expected.to eq(generate_metric_type) }
+  end
+
+  describe '#update_value' do
+    it 'increment values' do
+      expect(metric.values).to be_nil
+      metric.update_value(1)
+      expect(metric.values).to_not be_nil
+      expect(metric.values[0][1]).to eq 1
```

Comment:
Minor: This assertion is kinda not needed, since the next one can only pass if this one passes (same note for others in the file)

```suggestion
      expect(metric.values[0][1]).to eq 1
```